09/03/2018 21:12:07.0581	[raft_python_callback.py write_to_logger 68]  581 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:init_state_functions_handler, LINE:28
MESSAGE:start
-----------------------------

09/03/2018 21:12:07.0582	[raft_python_callback.py write_to_logger 68]  582 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:init_state_functions_handler, LINE:70
MESSAGE:end
-----------------------------

09/03/2018 21:12:07.0582	[raft_python_callback.py write_to_logger 68]  582 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:init_multicast_message, LINE:81
MESSAGE:init multicast socket
-----------------------------

09/03/2018 21:12:07.0582	[raft_python_callback.py write_to_logger 68]  582 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:init_multicast_message, LINE:88
MESSAGE:multicast was opened successfully
-----------------------------

09/03/2018 21:12:07.0582	[raft_python_callback.py write_to_logger 68]  582 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:13
MESSAGE:first time
-----------------------------

09/03/2018 21:12:07.0582	[raft_python_callback.py write_to_logger 68]  582 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 
-----------------------------

09/03/2018 21:12:07.0583	[raft_python_callback.py write_to_logger 68]  583 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-3 
-----------------------------

09/03/2018 21:12:07.0584	[raft_python_callback.py write_to_logger 68]  584 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:init_raft, LINE:189
MESSAGE:init succeed
-----------------------------

09/03/2018 21:12:07.0584	[raft_python_callback.py write_to_logger 68]  584 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:230
MESSAGE:start raft...
VARIABLES:  raft_ip=230.1.1.10 
-----------------------------

09/03/2018 21:12:07.0584	[raft_python_callback.py write_to_logger 68]  584 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:232
MESSAGE:start raft...
VARIABLES:  raft_port=12345  server_id=1  members_num=2  leader_timeout=50 
-----------------------------

09/03/2018 21:12:07.0584	[raft_python_callback.py write_to_logger 68]  584 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_raft, LINE:238
MESSAGE:run raft...
-----------------------------

09/03/2018 21:12:07.0584	[raft_python_callback.py write_to_logger 68]  584 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:131
MESSAGE:try to read message
-----------------------------

09/03/2018 21:12:10.0422	[raft_python_callback.py write_to_logger 68]  422 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:50
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=0 
-----------------------------

09/03/2018 21:12:10.0422	[raft_python_callback.py write_to_logger 68]  422 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=0 
-----------------------------

09/03/2018 21:12:13.0260	[raft_python_callback.py write_to_logger 68]  260 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:raft_timer, LINE:50
MESSAGE:add new timeout event- now in queue
VARIABLES:  sharedRaftData.raft_state.current_state=0 
-----------------------------

09/03/2018 21:12:13.0260	[raft_python_callback.py write_to_logger 68]  260 root [DEBUG] 
FILE:./C/Code/state_machine.c, FUNCTION:operate_machine_state, LINE:8
MESSAGE:enter state function
VARIABLES:  node->event=0  sharedRaftData.raft_state.current_state=0 
-----------------------------

09/03/2018 21:12:13.0261	[raft_python_callback.py write_to_logger 68]  261 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=term 
-----------------------------

09/03/2018 21:12:13.0261	[raft_python_callback.py write_to_logger 68]  261 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 
-----------------------------

09/03/2018 21:12:13.0263	[raft_python_callback.py write_to_logger 68]  263 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=status 
-----------------------------

09/03/2018 21:12:13.0263	[raft_python_callback.py write_to_logger 68]  263 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=-2 
-----------------------------

09/03/2018 21:12:13.0264	[raft_python_callback.py write_to_logger 68]  264 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:16
MESSAGE:update DB
VARIABLES:  DB_flag=status  key=leader_id 
-----------------------------

09/03/2018 21:12:13.0264	[raft_python_callback.py write_to_logger 68]  264 root [DEBUG] 
FILE:./C/Code/handlers_utils.c, FUNCTION:update_DB, LINE:17
MESSAGE:update DB
VARIABLES:  value=0 
-----------------------------

09/03/2018 21:12:13.0266	[raft_python_callback.py write_to_logger 68]  266 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:create_new_queue_node_data, LINE:107
MESSAGE:create new msg
VARIABLES:  event=8 
-----------------------------

09/03/2018 21:12:13.0266	[raft_python_callback.py write_to_logger 68]  266 root [DEBUG] 
FILE:./C/Code/follower_handlers.c, FUNCTION:follower_time_out_handler, LINE:254
MESSAGE:follower became candidate and is sending request for vote msg
-----------------------------

09/03/2018 21:12:13.0266	[raft_python_callback.py write_to_logger 68]  266 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:100
MESSAGE:try to send msg
VARIABLES:  message_size=20 
-----------------------------

09/03/2018 21:12:13.0266	[raft_python_callback.py write_to_logger 68]  266 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:send_raft_message, LINE:118
MESSAGE:msg was successfully sent
VARIABLES:  message_size=20 
-----------------------------

09/03/2018 21:12:13.0266	[raft_python_callback.py write_to_logger 68]  266 root [DEBUG] 
FILE:./C/Code/static_queue.c, FUNCTION:init_raft_queue, LINE:23
MESSAGE:not first time
-----------------------------

09/03/2018 21:12:13.0267	[raft_python_callback.py write_to_logger 68]  267 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:155
MESSAGE:msg was successfully read
VARIABLES:  message_size=20 
-----------------------------

09/03/2018 21:12:13.0267	[raft_python_callback.py write_to_logger 68]  267 root [DEBUG] 
FILE:./raft_main.c, FUNCTION:run_multicast_listener, LINE:21
MESSAGE:got new message and ids as followed:
VARIABLES:  new_node.message_sent_by=1  sharedRaftData.raft_state.server_id=1 
-----------------------------

09/03/2018 21:12:13.0267	[raft_python_callback.py write_to_logger 68]  267 root [DEBUG] 
FILE:./C/Code/raft_server.c, FUNCTION:get_raft_message, LINE:131
MESSAGE:try to read message
-----------------------------

